UOP1_Project/Assets/Scripts/CameraSwapTrigger.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Cinemachine;
+
+public class CameraSwapTrigger : MonoBehaviour
+{
+    public CinemachineVirtualCamera cameraToSwap;
+    
+    [Header("Asset references")]
+    [SerializeField] private VcamEventChannelSO VcamEventChannel = default;
+
+    BoxCollider boxCollider;
+
+    void Start()
+    {
+        boxCollider = GetComponent<BoxCollider>();
+    }
+
+
+   void OnTriggerExit(Collider other)
+   {
+        // check if it is the player
+        if(other.CompareTag("Player")){
+            
+            //check which side the player exited. we consider only the z axis so the enter and exit faces should be along the local z axis.
+            
+            Vector3 playerPosInLocalCoordinates = transform.InverseTransformPoint(other.transform.position);
+            
+            if(boxCollider.center.z > playerPosInLocalCoordinates.z){
+                
+                //player exited on -ve z side which means we switch to the default camera
+                VcamEventChannel.OnEventRaised(null);
+
+            }
+            else{
+
+                //player exited on +ve z side which means we switch to our camera
+                VcamEventChannel.OnEventRaised(cameraToSwap);
+            
+            }
+        }
+   }
+
+}
```

Comment:
I feel like this is veery hardcoded. You're thinking of the "entering the trunk" scenario, but what if we just want a special camera for the centre of a very open area? In that case you could enter and exit from all sides.

Instead of doing this, why don't we use the idea of Compound colliders? https://docs.unity3d.com/Manual/CollidersOverview.html?_ga=2.247206724.151196618.1613240916-822635934.1589716631

Reply:
So in case of compund colliders should we use trigger entry and exit to decide which camera to use rather than which side player exits on ?

Reply:
So in case of compound collider should we make it such that if the player is in the volume enclosed by the collider then we activate a special camera and when he exits that volume we go back to the default camera? Or do we keep the same behaviour as now but just consider the z axis of the particular child collider of the compound collider the player exits?

Reply:
Or I could setup the prefab to switch between both behaviours and the level designer can pick

Reply:
compound colliders implemented.